openlibrary/templates/home/index.html

```diff
@@ -37,3 +37,149 @@
+                return callback(result);
+            }
+        });
+    };
+
+    var recGetPopularEditions = function(all_books, available, waitlist, num_slots, start, limit, callback) {
+        if (available.length >= num_slots) {
+            return callback(all_books, available, waitlist);
+        } else {
+            getPopularEditionIds(start, limit, function(editions) {
+                if (editions.error) {
+                    console.log(editions.error);
+                    return callback(available, waitlist);
+                }
+                var batch = editions.books;
+                shuffle(batch);
+
+                // maintain a list of all books we know about, lookup
+                // table for later
+                for (var b in batch) {
+                    var edition = batch[b];
+                    var ocaid = edition[0];
+                    var ol_key = edition[1];
+                    all_books[ocaid] = ol_key;
+                }
+
+                var ocaids = batch.map(function(e) { return e[0] });
+                getAvailability(ocaids, function(responses) {
+                    var books = responses;
+                    start += limit;
+                    for (var b in books) {
+                        var book = books[b];
+                        if (book.status === "available") {
+                            available.push(b);
+                        } else if (book.status === "unavailable" && waitlist.length < num_slots) {
+                            waitlist.push(b);
+                        }
+                        if (available.length >= num_slots) {
+                            return callback(available, waitlist);
+                        }
+                    }
+                    return recGetPopularEditions(all_books, available, waitlist, num_slots, start, limit, callback);
+                });
+            });
+        }
+    }
+
+    function load_covers(carousel, state) {
+        for (var i = carousel.first; i <= carousel.last; i++) {
+            if (carousel.has(i)) {
+                carousel.get(i).find(".coverMagic").removeClass("hiddenCoverMagic");
+
+                var cover = carousel.get(i).find("img.cover");
+                if (cover.length > 0) {
+                    cover.attr("src", cover.attr("xsrc"));
+                }
+            }
+        }
+    }
+
+    var updateCarousel = function(cssid, books) {
+        var books = books.books;
+
+        \$('#' + cssid).empty();
+        for (var b in books) {
+            \$('#' + cssid).append('<li>' + books[b] + '</li>');
+        }
+
+        \$('#' + cssid + ' li').each(function(index, elem) {
+            \$(elem).find(".coverMagic").removeClass("hiddenCoverMagic");
+            var cover = \$(elem).find("img.cover");
+            if (cover.length > 0) {
+                cover.attr("src", cover.attr("xsrc"));
+            }
+        });
+
+        \$("#" + cssid).jcarousel({
+            scroll:6,
```

Comment:
This is the constructor to jcarousel, it specifies scrolling should be done 6 books at a time (e.g. a page at a time). Added comment to code